### PR TITLE
Db models

### PR DESCRIPTION
**Pull request by wrandall22**
This includes the database models for both KMA-3 and KMA-4

### CODE REVIEW

**Comment by twinge** [comment]
I love that you already have Travis integration on this. Good job on that.

File: db/migrate/20180312163100-create-score.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
+'use strict'
+module.exports = {
+  up: (queryInterface, Sequelize) => {
+    return queryInterface.createTable('scores', {
+      uri: {
```
Comment (Omicron7) [inline db/migrate/20180312163100-create-score.js]: I believe postgresql's default collation is case sensitive. We may want to enable case insensitive for uri's, or downcase everything before storing/querying.

**Comment by wrandall22** [comment]
Let me know if you want me to squash the fix commits

File: models/db/api-key.model.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
+'use strict'
+
+module.exports = (sequelize, DataTypes) => {
+  const apiKey = sequelize.define('Api_Key', {
```
Comment (Omicron7) [inline models/db/api-key.model.js]: Use `ApiKey` as the model name to match other class names.

File: models/db/score.model.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
+'use strict'
+
+module.exports = (sequelize, DataTypes) => {
+  const Score = sequelize.define('Score', {
+    uri: DataTypes.STRING,
+    unaware: DataTypes.INTEGER,
```
Comment (Omicron7) [inline models/db/score.model.js]: Sequelize supports automatic validation for columns. We should use that the integer fields and confidence.
http://docs.sequelizejs.com/manual/tutorial/models-definition.html#validations

File: models/db/score.model.js [hunk 4]
```diff
@@ -0,0 +1,16 @@
+'use strict'
+
+module.exports = (sequelize, DataTypes) => {
+  const Score = sequelize.define('Score', {
+    uri: DataTypes.STRING,
```
Comment (Omicron7) [inline models/db/score.model.js]: You could add a custom setter for uri that downcases the uri before storing, if we can't set collation to case insensitive.
http://docs.sequelizejs.com/manual/tutorial/models-definition.html#getters-setters

File: db/migrate/20180312163100-create-score.js [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+        defaultValue: 1
+      },
+      guide: {
+        type: Sequelize.INTEGER,
+        defaultValue: 1
+      },
+      confidence: {
+        type: Sequelize.DOUBLE,
+        defaultValue: 0
+      }
```
Comment (Omicron7) [inline db/migrate/20180312163100-create-score.js]: I believe `created_at` and `updated_at` need to be defined in the migration unless `timestamps: false` is defined in the model options

File: db/migrate/20180312163100-create-score.js [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+    return queryInterface.createTable('scores', {
+      uri: {
+        allowNull: false,
+        primaryKey: true,
+        type: Sequelize.STRING
+      },
+      unaware: {
+        type: Sequelize.INTEGER,
+        defaultValue: 1,
+        validate: {
```
Comment (Omicron7) [inline db/migrate/20180312163100-create-score.js]: `validate` should should only be on the model definition (models/db/*.model.js) and not the migration. It probably wont error here, but I don't think it adds any sort of database constraint.

File: models/db/api-key.model.js [hunk 7]
```diff
@@ -0,0 +1,14 @@
+'use strict'
+
+module.exports = (sequelize, DataTypes) => {
+  const ApiKey = sequelize.define('Api_Key', {
```
Comment (Omicron7) [inline models/db/api-key.model.js]: The string `'Api_Key'` should be `'ApiKey'`. This is the value index.js uses to create the class name of the model.

File: models/db/score.model.js [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+      },
+      set (val) {
+        this.setDataValue('uri', val.toLowerCase())
+      }
+    },
+    unaware: {
+      type: DataTypes.INTEGER,
+      validate: {
+        min: 1,
+        max: 4
```
Comment (Omicron7) [inline models/db/score.model.js]: `max:` should be 5 for unaware, curious, follower, and guide.

File: models/db/score.model.js [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+
+module.exports = (sequelize, DataTypes) => {
+  const Score = sequelize.define('Score', {
+    uri: {
+      type: DataTypes.STRING,
+      get () {
+        return this.getDataValue('uri').toLowerCase()
+      },
+      set (val) {
+        this.setDataValue('uri', val.toLowerCase())
```
Comment (Omicron7) [inline models/db/score.model.js]: I'm not sure if the setter is called before or after validations. You may need to check for null or undefined value before calling toLowerCase()

**Comment by wrandall22** [comment]
Thanks for reviewing!

**Pull request merged**